### DISCUSSION TITLE
Synchronized Desktop Locking

### DISCUSSION THREAD

**Discussion by ghost** [input-leap/input-leap#1748] [Feature Requests]
> This issue has been migrated from old Barrier Github repository https://github.com/debauchee/barrier/issues/177
> 
> Issue created on: 2018-11-17 by @DragoonAethis
> Issue last updated on: 2020-07-09

### Operating Systems/Barrier Version ###

- Barrier: 2.1.0 (everywhere)
- Server: Arch (w/ KDE 5.14)
- Client: Windows 10 (1803)

### Feature request ###

Currently there are some keystrokes that are not possible to easily emulate under most platforms. One of these is Win+L which locks the Windows session, others would emulate keystrokes like Ctrl-Alt-L to lock Gnome, KDE sessions (and sometimes, emulated keystrokes aren't caught). This is a [known bug for years under Synergy](https://github.com/symless/synergy-core/issues/3084).

A possible solution would be to implement OS-specific desktop locking functions (eg running `loginctl lock-session` under Linux, calling `LockWorkStation` on user32.dll under Windows, calling `SACLockScreenImmediate` from the `login` framework under macOS), and have these available in a portable manner as an action (so that one could add something like `keystroke(ctrl+alt+l) = lockScreen(*)` to the config and have all/chosen/current screen locked).

---

> Commented on: 2018-11-30 by @a7hybnj2

would be great to have this working. Even my 'sync screen savers' isn't working. I read it might have to do with not using xscreensaver on my linux server but even trying that did not produce results.

---

> Commented on: 2020-04-23 by @GeertGeurts

This would be a great feature. Currently I'm working with two laptops, a linux client next to a windows server. When I'm working on the windows laptop for too long, and I see my linux screen fading out(lock screen goes in a fading fasion...), then I can prevent it from locking by quickly moving my mouse over to the linux client. Would be great to have a feature that prevents the need for this action!

---

> Commented on: 2020-07-09 by @imsys

I found about Berrier today, installed the server on my Linux laptop, and used the Windows PC as a client.
As my laptop use less energy and doesn't produce noise, I usually don't power it off, so I decided to set it as the server.
I let most of my security sensitive stuff on my Linux and I like to let the screen locked when not working on it, but with Berrier doesn't work with my Linux screen locked.
I tried with in my Arch XFCE desktop, with light-locker, xfce4-screensaver, xscreensaver, gnome-screensaver, slock and xlockmore. All of those did not let me move the mouse to the Windows Client.

I wonder what could be done to allow it: I wonder if it was directly restricted in Barrier, or if it is the screenlockers softwares' restriction.

---

> Commented on: 2020-07-09 by @DragoonAethis

I'd say that the point of a screen locker is to prevent any sort of interaction with a host until you've unlocked it :P At least on  Linux + X11, screen lockers will grab all input and prevent any background apps (such as Barrier) from handling it.

**Comment by wokawoka** [comment]
any news on this?

**Comment by shymega** [comment]
> any news on this?

Whilst I understand why you ask this, and I do the same thing on some repos too, generally, I think it's best to assume 'no news' **means** 'no news'. This project relies on voluntary contributions, we're not sponsored, which means we don't get full-time devs working on it.

I'm going to go ahead and 'hide' your comment to tidy up the issue.
